Mathlib/Init/Set.lean

```diff
@@ -102,6 +102,40 @@ for instance, `{(x) | (x : Nat) (y : Nat) (_hxy : x = y^2)}`.
 macro (priority := low) "{" t:term " | " bs:extBinders "}" : term =>
   `({x | ∃ᵉ $bs:extBinders, $t = x})
 
+/-- Pattern matching in set-builder notation
+If `pat` is a pattern that is matched by all objects of type `X`, then `{ pat : X | p }` is
+notation for the set of all objects of type `X` which, when matched with the pattern `pat`,
+make `p` come out true.  If `X` can be inferred, then `{ pat | p }` can be used.
```

Comment:
I made a small change ("objects of type `X`" rather than "elements in `X`").  (I know many people think of types as being essentially the same as sets, but I find it helpful to keep the ideas separate.  I thought it might be better to reserve the word "element" for sets.)

When I wrote the docstring, I didn't realize that I was writing a docstring--I thought I was just writing a comment for anyone who looked at the file.  Thanks for explaining how docstrings work.